console/console-init/ui/src/modules/iot-device/components/CreateMetadata/MetaDataRow.tsx

```diff
@@ -29,16 +29,43 @@ export const MetaDataRow: React.FC<IMetaDataRow> = ({
   rowIndex
 }) => {
   const metadataRow = metadataList[rowIndex];
-  // console.log("metadataList", metadataList, "type", metadataRow.type);
+  const [validationStatus, setValidationStatus] = useState<"default" | "error">(
+    "default"
+  );
 
   const onSelectType = (typeValue: string) => {
+    const validationStatus = getValidationStatus(typeValue, metadataRow.value);
+    setValidationStatus(validationStatus);
     let updatedTypeMetadata = [...metadataList];
     updatedTypeMetadata[rowIndex].type = typeValue;
     if (isObjectOrArray(typeValue as any))
       updatedTypeMetadata[rowIndex].value = "";
     setMetadataList(updatedTypeMetadata);
   };
 
+  const getValidationStatus = (type: string, value: string) => {
+    let validationStatus: "default" | "error" = "default";
+    switch (type) {
+      case "string":
+        validationStatus = typeof value === "string" ? "default" : "error";
```

Comment:
A query here, if we want to put a number having the metadata type as string, it is still validating it, is that expected?

Reply:
I think in string we should skip validation. we can just do empty string check as everything in input value can be a string

Reply:
@drcoolsanjeev The check for string is needed here because if user first chose the type as number by mistake and entered a string as value, there would be a validation error shown on the screen. With this check in place, the validation error would be removed if the user changes the type back to string. Although, I agree that we should have an empty string check somewhere.